spec/models/spree/permission_sets/supplier/staff_ability_spec.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'cancan'
+require 'cancan/matchers'
+require 'spree/testing_support/ability_helpers'
+
+describe Spree::PermissionSets::Supplier::AdminAbility do
+  let(:ability) { Spree::Ability.new(user) }
+  let(:supplier) { create(:supplier) }
+  let(:supplier_staff_role) { build(:role, name: 'supplier_staff') }
+  let(:user) { create(:user, supplier: supplier) }
+  let(:token) { nil }
+
+  subject { ability }
+
+  before(:each) do
+    user.spree_roles << supplier_staff_role
+    described_class.new(ability).activate!
+  end
+
+  context 'for Product' do
+    let(:resource) { create(:product) }
+
+    before(:each) do
+      resource.add_supplier!(user.supplier)
+      resource.reload
+    end
+
+    context 'requested by another suppliers user' do
+      let(:other_resource) {
+        product = create(:product)
+        product.add_supplier!(create(:supplier))
+        product
+      }
+
+      it { expect(ability).to_not be_able_to :read, other_resource }
+      it { expect(ability).to_not be_able_to :admin, other_resource }
+      it { expect(ability).to_not be_able_to :edit, other_resource }
+    end
+
+    context 'requested by suppliers user' do
+      let(:resource) {
+        product = create(:product)
+        product.add_supplier!(user.supplier)
+        product.reload
+      }
```

Comment:
This seems a bit weird for me. I think could be better something like
```suggestion
      let(:resource) { create(:product) }
      before do
        product.add_supplier!(user.supplier)
        product.reload
      end
```

Reply:
done ^^ thanks